tests/if/if7.dora

```diff
@@ -0,0 +1,14 @@
+fun main() {
+  let x = if "abc"
+    ... == "bcd" { 1 }
+    ... != "abc" { 2 }
+    ... .size() == 3L { 3 }
+    else { -1 };
+  assert(x == 3);
+  let x = if "abc"
```

Comment:
I should add an test where the cond_head is a function call to show that the hoisting into a let works as expected and the function is only executed once.

Reply:
Done.